packages/opds-web-client/src/components/context/__tests__/StoreContextTest.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from "react";
+import { expect } from "chai";
+import { stub } from "sinon";
+
+import * as PropTypes from "prop-types";
+import { shallow, mount } from "enzyme";
+import { ReactReduxContext } from "react-redux";
+
+import buildStore from "../../../store";
+
+import StoreContextProvider from "../StoreContext";
+import PathForProvider from "../PathForContext";
+import { PathFor } from "../../../interfaces";
+import { isNull } from "util";
+// import { mockRouterContext } from "../../../routing";
+
+/**
+ *  Testing the StoreContext component. Based on old OPDSCatalog Tests
+ *    - if not given state, it will build state on its own
+ *      (seems like this should be a test of buildState maybe?)
+ *    - if given a state, it will use it and pass it down.
+ *    - passes props down
+ *    - test that the context really is available through old and new APIs (and it's the same?)
+ *
+ *    * pathFor must be available in context. Use the pathForProvider to wrap tests
+ */
+
+const pathFor: PathFor = (collectionUrl, bookUrl) => {
+  return "path";
+};
+
+const PathForContext = ({ children }) => {
+  return <PathForProvider pathFor={pathFor}>{children}</PathForProvider>;
+};
+
+describe("StoreContext", () => {
+  const props = {
+    initialState: null,
+    authPlugins: null
+  };
+
+  class Child extends React.Component {
+    static contextType = ReactReduxContext;
+    render() {
+      return <div>I should have access to context</div>;
+    }
+  }
+
+  // provide it as a wrapping component to the test
+  const CombinedContext = ({ children, ...otherProps }) => (
+    <PathForContext>
+      <StoreContextProvider {...props} {...otherProps}>
+        {children}
+      </StoreContextProvider>
+    </PathForContext>
+  );
+
+  it("passes store down via new API", () => {
+    let wrapper = mount(<Child />, { wrappingComponent: CombinedContext });
+
+    // now we need to see if child has access to context via new api
+    expect(wrapper.context().store).to.be.ok;
+    expect(wrapper.context().store).to.have.property("dispatch");
+  });
+
+  it("creates a new store if not given a state", () => {
+    let wrapper = mount(<Child />, { wrappingComponent: CombinedContext });
+
+    // now we need to see if child has access to context via new api
+    expect(wrapper.context().store).to.be.ok;
+    expect(wrapper.context().store).to.have.property("dispatch");
+  });
```

Comment:
This test is the same as the previous test.

Reply:
That's true. I actually did that intentionally because it is actually two different things that need to be tested and one test might change from the other over time. I can make them one test with a longer title if you prefer though

Reply:
I see, or maybe instead of one test add a comment to explain why they are the same.